### DISCUSSION TITLE
Spec compliance feasability

### DISCUSSION THREAD

**Discussion by lastmjs** [CanadaHonk/porffor#10] [General]
Will it be possible for porffor to ever be fully ECMAScript compliant? I'm not asking if the team can or will make time, but if it's theoretically possible to compile any and all ECMAScript-compliant ES2023 to Wasm, for example.

Well maybe that's just my first question. Then my second question would be if that's the eventual end-goal if development continues.

**Comment by CanadaHonk** [comment]
Good question! I think yes, eventually, but with some tradeoffs. Essentially, the more you are spec compliant in this case the more potential performance you can lose.

My plan with Porffor is to try and be more spec compliant (especially more now than in the past), and have many prefs allowing developers to fine tune what spec compliance they need and don't need to try and maximize performance.

(Also  "the team" is me in my free time and that is it :grin:)

I'll leave this issue open as I think it is a good and important general discussion.

**Comment by erlend-sh** [comment]
For frontend use, high compatibility with [Web Components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components) (and by extension, frameworks like [Lit](https://lit.dev/)) might be a fitting goal?
